database/plugin/metadata/sqlite/utxo.go

```diff
@@ -203,6 +205,7 @@ func (d *MetadataStoreSqlite) SetUtxo(
 	payment []byte, // payment
 	stake []byte, // stake
 	amount uint64, // amount
+	asset *lcommon.MultiAsset[lcommon.MultiAssetTypeOutput], // asset
```

Comment:
This should be assets (it's multi-asset, so multiple). Add them to tmpUtxo.Assets before saving tmpUtxo if it's not nil.